client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java

```diff
@@ -202,4 +206,97 @@ public String toString() {
+      properties.setProperty(ConfigurationProperties.MARSHALLER, marshallerClass().getName());
+      properties.setProperty(ConfigurationProperties.PROTOCOL_VERSION, protocolVersion());
+      properties.setProperty(ConfigurationProperties.SO_TIMEOUT, Integer.toString(socketTimeout()));
+      properties.setProperty(ConfigurationProperties.TCP_NO_DELAY, Boolean.toString(tcpNoDelay()));
+      properties.setProperty(ConfigurationProperties.TCP_KEEP_ALIVE, Boolean.toString(tcpKeepAlive()));
+      properties.setProperty(ConfigurationProperties.TRANSPORT_FACTORY, transportFactory().getName());
+      properties.setProperty(ConfigurationProperties.VALUE_SIZE_ESTIMATE, Integer.toString(valueSizeEstimate()));
+      properties.setProperty(ConfigurationProperties.MAX_RETRIES, Integer.toString(maxRetries()));
+
+      properties.setProperty("exhaustedAction", Integer.toString(connectionPool().exhaustedAction().ordinal()));
+      properties.setProperty("maxActive", Integer.toString(connectionPool().maxActive()));
+      properties.setProperty("maxTotal", Integer.toString(connectionPool().maxTotal()));
+      properties.setProperty("maxWait", Long.toString(connectionPool().maxWait()));
+      properties.setProperty("maxIdle", Integer.toString(connectionPool().maxIdle()));
+      properties.setProperty("minIdle", Integer.toString(connectionPool().minIdle()));
+      properties.setProperty("numTestsPerEvictionRun", Integer.toString(connectionPool().numTestsPerEvictionRun()));
+      properties.setProperty("minEvictableIdleTimeMillis", Long.toString(connectionPool().minEvictableIdleTime()));
+      properties.setProperty("timeBetweenEvictionRunsMillis", Long.toString(connectionPool().timeBetweenEvictionRuns()));
+
+      properties.setProperty("lifo", Boolean.toString(connectionPool().lifo()));
+      properties.setProperty("testOnBorrow", Boolean.toString(connectionPool().testOnBorrow()));
+      properties.setProperty("testOnReturn", Boolean.toString(connectionPool().testOnReturn()));
+      properties.setProperty("testWhileIdle", Boolean.toString(connectionPool().testWhileIdle()));
+
+      StringBuilder servers = new StringBuilder();
+      for(ServerConfiguration server : servers()) {
+         if (servers.length() > 0) {
+            servers.append(";");
+         }
+         servers.append(server.host()).append(":").append(server.port());
+      }
+      properties.setProperty(ConfigurationProperties.SERVER_LIST, servers.toString());
+
+      properties.setProperty(ConfigurationProperties.USE_SSL, Boolean.toString(security.ssl().enabled()));
+
+      if (security.ssl().keyStoreFileName() != null)
+         properties.setProperty(ConfigurationProperties.KEY_STORE_FILE_NAME, security.ssl().keyStoreFileName());
+
+      if (security.ssl().keyStorePassword() != null)
+         properties.setProperty(ConfigurationProperties.KEY_STORE_PASSWORD, new String(security.ssl().keyStorePassword()));
+
```

Comment:
I need to rectify my statement. Can you also make the callbackHandler with both with a String (and therefore initialize it with reflection) or an instance ?